FRAuth/FRAuth/Config/OAuth2Client.swift

```diff
@@ -458,4 +462,28 @@ public class OAuth2Client: NSObject, Codable {
         //  Call /token service to exchange auth code to OAuth token set
         return Request(url: self.serverConfig.tokenURL, method: .POST, headers: header, bodyParams: parameter, requestType: .urlEncoded, responseType: .json, timeoutInterval: self.serverConfig.timeout)
     }
+
+    /// Builds /endSession request for an external user-agent based on given OAuth2 client information
+    /// - Parameters:
+    ///   - idToken: OIDC id_token
+    /// - Returns: Request object
+    func buildEndSessionRequestForExternalAgent(idToken: String?) -> Request {
+      //  Construct parameter for the request
+      var parameter: [String: String] = [:]
+      parameter[OAuth2.postLogoutRedirectUri] = self.signoutRredirectUri!.absoluteString
```

Comment:
Do not use force unwrap on self.signoutRredirectUri!.absoluteString, as they might use a bad URL or this is optional as well

Reply:
fixed